dev/import-beats/kibana.go

```diff
@@ -0,0 +1,220 @@
+	}
+)
+
+type kibanaContent struct {
+	dashboardFiles     map[string][]byte
+	visualizationFiles map[string][]byte
+}
+
+type kibanaMigrator struct {
+	hostPort string
+}
+
+type kibanaDocuments struct {
+	Objects []mapStr `json:"objects"`
+}
+
+func newKibanaMigrator(hostPort string) *kibanaMigrator {
+	return &kibanaMigrator{
+		hostPort: hostPort,
+	}
+}
+
+func (km *kibanaMigrator) migrateDashboardFile(dashboardFile []byte) ([]byte, error) {
+	dashboardFile, err := prepareDashboardFile(dashboardFile)
+	if err != nil {
+		return nil, errors.Wrapf(err, "preparing file failed")
+	}
+
+	request, err := http.NewRequest("POST",
+		fmt.Sprintf("http://%s/api/kibana/dashboards/import?force=true", km.hostPort),
+		bytes.NewReader(dashboardFile))
+	if err != nil {
+		return nil, errors.Wrapf(err, "creating POST request failed")
+	}
+	request.Header.Add("kbn-xsrf", "8.0.0")
+	response, err := http.DefaultClient.Do(request)
+	if err != nil {
+		return nil, errors.Wrapf(err, "making POST request to Kibana failed")
+	}
+	defer response.Body.Close()
+
+	saved, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, errors.Wrapf(err, "reading saved object failed")
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("making POST request failed: %s", string(saved))
+	}
+	return saved, nil
+}
+
+func prepareDashboardFile(dashboardFile []byte) ([]byte, error) {
+	var documents kibanaDocuments
+
+	err := json.Unmarshal(dashboardFile, &documents)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unmarshalling dashboard file failed")
+	}
+
+	for i, object := range documents.Objects {
+		object, err = encodeFields(object)
+		if err != nil {
+			return nil, errors.Wrapf(err, "encoding fields failed")
+		}
+		documents.Objects[i] = object
+	}
+
+	data, err := json.Marshal(&documents)
+	if err != nil {
+		return nil, errors.Wrapf(err, "marshalling dashboard file failed")
+	}
+	return data, nil
+}
+
+func encodeFields(ms mapStr) (mapStr, error) {
+	for _, field := range encodedFields {
+
+		v, err := ms.getValue(field)
+		if err == errKeyNotFound {
+			continue
+		} else if err != nil {
+			return mapStr{}, errors.Wrapf(err, "retrieving value failed (key: %s)", field)
+		}
+
+		ve, err := json.Marshal(v)
+		if err != nil {
+			return mapStr{}, errors.Wrapf(err, "marshalling value failed (key: %s)", field)
+		}
+
+		_, err = ms.put(field, string(ve))
+		if err != nil {
+			return mapStr{}, errors.Wrapf(err, "putting value failed (key: %s)", field)
+		}
+	}
+	return ms, nil
+}
+
+func createKibanaContent(kibanaMigrator *kibanaMigrator, modulePath string) (kibanaContent, error) {
+	moduleDashboardPath := path.Join(modulePath, "_meta", "kibana", "7", "dashboard")
+	moduleDashboards, err := ioutil.ReadDir(moduleDashboardPath)
+	if os.IsNotExist(err) {
+		log.Printf("\tno dashboards present, skipped (modulePath: %s)", modulePath)
+		return kibanaContent{}, nil
+	} else if err != nil {
```

Comment:
Nit: Problably no else needed as you return in the if clause above.

Reply:
Fixed.